[4.0] No user information if the user hast not the right to manage users

chmst opened the pull request:
Pull Request for Issue #29589 .

### Summary of Changes
In general, the site information has view level "super users". So in geeral, module "site information" is not shown for users of other groups.
But this view level can be changed to public or special.

If a user hast not permission to manage users, he should not see the users information.


### Testing Instructions
Make sure that the admin module "site information" has view level "manage" or lower.
Login as a user in a group without the right to manage users. 
You may not see the users information. 

### Expected result

<img width="514" alt="site-information" src="https://user-images.githubusercontent.com/1035262/84568472-09cbed00-ad80-11ea-8253-9cc1cb046aeb.PNG">

### Remark
Users and articles information are shown in many places. 
I think this information could be removed from site information as well.

chmst committed:
No user information if the user hast not the right to manage users

Quy commented on administrator/modules/mod_stats_admin/src/Helper/StatsAdminHelper.php:
```diff
@@ -103,7 +105,7 @@ public static function getStats(Registry $params, CMSApplication $app, DatabaseI
 			}
 
-			if ($users)
+			if ($users && $user->authorise('core.PR an', 'com_users'))
```
```suggestion
			if ($users && $user->authorise('core.manage', 'com_users'))
```

richard67 replied:
@chmst Sure that's resolved?

particthistle replied:
Changing it to core.manage at line 108 then resolved the issue I had with the PR test.

Further to @brianteeman comment, changing 108-114 to the following would disable the link, but still display the number of users. It just doesn't format the number as it's no longer a link.

```
			if ($users)
			{
				$rows[$i]        = new \stdClass;
				$rows[$i]->title = Text::_('MOD_STATS_USERS');
				$rows[$i]->icon  = 'users';
				$rows[$i]->data  = $users;
				if ($user->authorise('core.manage', 'com_users'))
				    {
				    $rows[$i]->link  = Route::_('index.php?option=com_users');
				    }
				$i++;
			}
```

particthistle replied:
Missed this when I did my testing as I'd had this on the screen for a few hours tonight and missed it.

particthistle commented:
I have tested this item :red_circle: unsuccessfully on 269d0b77e3440f0793d825cf0e9c15d44757d77f<br /><br />Partially tested successfully.

**Manager level user _Success_**
- Manager doesn't have User management access
- Site Information module set to "Special"
- Number of users on the site not shown.

**Administrator level user _Issue Discovered_**
- Administrator has User management access
- Site Information module set to "Special"
- Number of users on the site not shown, but as the Administrator has User Management access, they should be able to see the number of users?

Line 108 looks to be overzealous:

```if ($users && $user->authorise('core.PR an', 'com_users'))```

_Still working out Joomla function code... what's the 'core.PR an' permission do?_<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29596">issues.joomla.org/tracker/joomla-cms/29596</a>.</sub>

brianteeman commented:
> If a user hast not permission to manage users, he should not see the users information.

I do not agree with this assumption

I think it is perfectly valid that the site information module displays the count of users. It is only the link that needs to be removed

particthistle commented:
Changing it to core.manage at line 108 per @Quy suggestion resolved the issue I had with the PR test.

```
if ($users && $user->authorise('core.manage', 'com_users'))
```
@brianteeman I'm split on whether the non-user managers should be able to see the number of users.

Further to @brianteeman comment, changing 108-114 to the following would disable the link, but still display the number of users. It just doesn't format the number as it's no longer a link.

```
			if ($users)
			{
				$rows[$i]        = new \stdClass;
				$rows[$i]->title = Text::_('MOD_STATS_USERS');
				$rows[$i]->icon  = 'users';
				$rows[$i]->data  = $users;
				if ($user->authorise('core.manage', 'com_users'))
				    {
				    $rows[$i]->link  = Route::_('index.php?option=com_users');
				    }
				$i++;
			}
```
![image](https://user-images.githubusercontent.com/5515866/84574038-775f3400-ade7-11ea-8302-4947f8f1dab8.png)

chmst committed:
fix condition, hide only the link

chmst commented:
Thank you for testing and your opinion. I have fixed and changed as you prefer to remove only the link. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29596">issues.joomla.org/tracker/joomla-cms/29596</a>.</sub>

Quy commented:
I have tested this item :white_check_mark: successfully on ac3f59f52741dac1852a1d99e6852b69c01af205<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29596">issues.joomla.org/tracker/joomla-cms/29596</a>.</sub>

Quy commented on administrator/modules/mod_stats_admin/src/Helper/StatsAdminHelper.php:
```diff
@@ -109,7 +111,10 @@ public static function getStats(Registry $params, CMSApplication $app, DatabaseI
-				$rows[$i]->link  = Route::_('index.php?option=com_users');
+				if ($user->authorise('core.manage', 'com_users'))
+				{
+					$rows[$i]->link  = Route::_('index.php?option=com_users');
```
```suggestion
					$rows[$i]->link = Route::_('index.php?option=com_users');
```

zero-24 commented on administrator/modules/mod_stats_admin/src/Helper/StatsAdminHelper.php:
```diff
@@ -109,7 +111,10 @@ public static function getStats(Registry $params, CMSApplication $app, DatabaseI
 				$rows[$i]->icon  = 'users';
 				$rows[$i]->data  = $users;
-				$rows[$i]->link  = Route::_('index.php?option=com_users');
+				if ($user->authorise('core.manage', 'com_users'))
```
```suggestion

				if ($user->authorise('core.manage', 'com_users'))
```

Quy commented on administrator/modules/mod_stats_admin/src/Helper/StatsAdminHelper.php:
```diff
@@ -109,7 +111,10 @@ public static function getStats(Registry $params, CMSApplication $app, DatabaseI
+				{
+					$rows[$i]->link  = Route::_('index.php?option=com_users');
+				}
 				$i++;
```
```suggestion

				$i++;
```

particthistle commented:
I have tested this item :white_check_mark: successfully on ac3f59f52741dac1852a1d99e6852b69c01af205<br /><br />Revised version tested as expected.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29596">issues.joomla.org/tracker/joomla-cms/29596</a>.</sub>

chmst committed:
Update administrator/modules/mod_stats_admin/src/Helper/StatsAdminHelper.php

Co-authored-by: Tobias Zulauf <zero-24@users.noreply.github.com>

chmst committed:
Update administrator/modules/mod_stats_admin/src/Helper/StatsAdminHelper.php

Co-authored-by: Quy <quy@fluxbb.org>

chmst committed:
Update administrator/modules/mod_stats_admin/src/Helper/StatsAdminHelper.php

Co-authored-by: Quy <quy@fluxbb.org>

richard67 commented:
Last 3 commits after @particthistle 's test have just been code style, so his test result is still valid. I'll add it back in the issue tracker so it is correctly counted.

@Quy After your test there have been changes which were not code style only. Could you test again? Thanks in advance.

Quy commented:
I have tested this item :white_check_mark: successfully on 01c5665ecd59e7c571880f028290c85051934b9c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29596">issues.joomla.org/tracker/joomla-cms/29596</a>.</sub>

Quy commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29596">issues.joomla.org/tracker/joomla-cms/29596</a>.</sub>

richard67 commented:
@zero-24 GitHub still shows that there are changes requested. I think it's due to your past review, but it is solved meanwhile as far as I can see. Could you just review again so this display of requested changes disappears if all is ok?

wilsonge commented:
Thanks!
